### ISSUE TITLE
Add example thumbnails of `optuna.visualization` features

### ISSUE THREAD

**Issue by Alnusjaponica** [optuna/optuna#4944]
## Motivation
Enhance the document's visual appeal for better comprehension.

## Detail
The primary goal of this issue is to provide a comprehensive overview of examples for each visualization feature on the [visualization document top](https://optuna.readthedocs.io/en/stable/reference/visualization/index.html). The inclusion of visual examples significantly improves the understanding of the intended purpose for each plot function.

While this may require some discussion, one straightforward method to incorporate visual explanations is by adding thumbnails with links to examples, similar to [`matplotlib.pyplot.scatter`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.scatter.html#examples-using-matplotlib-pyplot-scatter):
<img width="808" alt="matplotlib_example" src="https://github.com/optuna/optuna/assets/50256998/04367e6b-b8ea-4253-aa79-4985f3c30097">

The challenge lies in avoiding code duplication when using [sphinx gallery](https://sphinx-gallery.github.io/stable/configuration.html#examples-using-numpy-exp). It is essential to note that generating figures automatically would be ideal to prevent an increase in maintenance costs associated with this change.

**Comment by Cupcha** [comment]
Hello ! We can work on it.

**Comment by Alnusjaponica** [comment]
I apologize for my delayed reply. We appreciate your proposal and eagerly anticipate your PR!

**Comment by GuillaumeFerreol** [comment]
Hello @Alnusjaponica, I am working with Cupcha on this issue. 
I saw that in the documentation there is already tutorial with visualization examples of different plot types. There are also a few examples on the documentation page of each plot type. 
What exactly did you have in mind ? For each plot type replace the example currently included by a thumbnail (like in matplotlib) linking to a specific tutorial (and include all the thumbnails on the `optuna.visualization` documentation) ?

**Comment by Alnusjaponica** [comment]
Thank you for your question. The objective of this issue is to enable a comprehensive overview of examples for each visualization feature. To achieve this, I suggest adding a list that includes thumbnail plots as a possible examples.
I apologize for any confusion, and the description has been updated accordingly.

As originally mentioned, the solution to address this issue is not straightforward, hence the "needs-discussion" label has been applied. While Sphinx Gallery appears to be the most straightforward example known at the moment, I've hesitated to start it because simply adding tutorial code may lead to code duplication, which is not ideal. If you have any suggestions for a better approach, I would greatly appreciate it.

**Comment by Alnusjaponica** [comment]
Certainly, it is indeed possible to arrange images with links promptly. However, when modifying functions within the 'visualization' module, it becomes necessary to update the images each time. We have concluded that this may lead to an undesirable increase in maintenance costs. We apologize for the inconvenience, but we have updated the description further based on your feedback. Thank you for your valuable input.
